### DISCUSSION TITLE
Piano roll:  How to prevent automatic cropping?

### DISCUSSION THREAD

**Discussion by randolf** [LMMS/lmms#6611] [Support request] [answered]
I've got a final note in my piano roll that has a slow release in the envelope, but it gets cut off early because of some automatic cropping.  This, of course, ruins the music.

How can I arbitrarily set the final [empty] segment?

Thanks in advance.

**Comment by allejok96** [comment, marked as answer]
If it's the final note of the song that gets cut off too early you can make a loop from beginning to end and select *Export between loop markers* when you save the song. 

Or if you're hacky I guess you could insert a quiet note at the end but I'm not sure...

**Reply by randolf** [reply]
Yes, the final notes are (e.g., a chord) are getting cut off at the end.  Loop markers also don't solve the problem, and I don't want to have to go to the extent of exporting many loops just for this -- it's counter-intuitive (although it is a solution so I marked your response as an answer since it probably will be helpful to people working on smaller projects where exporting as a loop should work as an interim solution).

A quiet note is, indeed, "hacky," because it will cause confusion when someone else edits or when I edit and forgot why I an extra note is appearing.

**Comment by randolf** [comment]
LMMS still has a problem though, because we really need a way to set the size of a piano roll area so that notes that have reverberation, echo, etc., that continue to play beyond the note don't get abruptly cut off.

(This can be done in other music composition software, such as Mixcraft, and pretty much all others that I've seen so far, including some software I used on the Commodore 64 back in the 1980s but don't remember its name.)
